src/taipy/core/config/core_section.py

```diff
@@ -93,6 +101,10 @@ def __init__(
         self.mode = mode or self._DEFAULT_MODE
         self.version_number = version_number or self._DEFAULT_VERSION_NUMBER
         self.force = force or self._DEFAULT_FORCE
+
+        self._core_version = core_version or self._CURRENT_CORE_VERSION
+        self._check_compatibility(self._core_version)
```

Comment:
If there is no version in a TOML file, we want to keep it `None` and the check_compatibility method should raise an error. 
No? 
We don't want to use the current version and then check the compatibility.

Reply:
then wouldn't that make all of current existing toml raises error? since we haven't introduce core_version up until now